src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -0,0 +1,338 @@
+    //   name relative to other names, given a specific search string.
+    //   Currently, this is based off of two factors:
+    //   * The weight is incremented once for each matched character of the
+    //     search text.
+    //   * If a matching character from the search text was found at the start
+    //     of a word in the name, then we increment the weight again.
+    //     * For example, for a search string "sp", we want "Split Pane" to
+    //       appear in the list before "Close Pane"
+    // Arguments:
+    // - searchText: the string of text to search for in `name`
+    // - name: the name to check
+    // Return Value:
+    // - the relative weight of this match
+    int CommandPalette::_getWeight(const winrt::hstring& searchText,
+                                   const winrt::hstring& name)
+    {
+        std::wstring lowercaseSearchText{ searchText.c_str() };
+        std::wstring lowercaseName{ name.c_str() };
+        std::transform(lowercaseSearchText.begin(), lowercaseSearchText.end(), lowercaseSearchText.begin(), std::towlower);
+        std::transform(lowercaseName.begin(), lowercaseName.end(), lowercaseName.begin(), std::towlower);
+
+        int totalWeight = 0;
+        bool lastCharWasSpace = true;
+        const wchar_t* namePtr = lowercaseName.c_str();
+        const wchar_t* endOfName = lowercaseName.c_str() + lowercaseName.size();
+        for (const auto& wch : lowercaseSearchText)
+        {
+            while (wch != *namePtr)
+            {
+                // increment the character to look at in the name string
+                namePtr++;
+                if (namePtr >= endOfName)
+                {
+                    // If we are at the end of the name string, we haven't found all the search chars
+                    return totalWeight;
+                }
+                lastCharWasSpace = *namePtr == L' ';
+            }
+
+            // We found the char, increment weight
+            totalWeight++;
+            totalWeight += lastCharWasSpace ? 1 : 0;
+        }
+        return totalWeight;
+    }
+
+    void CommandPalette::SetDispatch(const winrt::TerminalApp::ShortcutActionDispatch& dispatch)
+    {
+        _dispatch = dispatch;
+    }
+
+    // Method Description:
+    // - Dismiss the command palette. This will:
+    //   * select all the current text in the input box
+    //   * set our visibility to Hidden
+    //   * raise our Closed event, so the page can return focus to the active Terminal
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void CommandPalette::_close()
+    {
+        Visibility(Visibility::Collapsed);
+        // TODO: Do we want to clear the text box each time we close the dialog? Or leave it?
+        // I think if we decide to leave it, we should auto-select all the text
+        // in it, so a user can start typing right away, or continue with the
+        // current selection.
```

Comment:
do what search did

Reply:
For reference, VS Code saves the text for search, but clears it for Command Palette. That said, idk what to do. I'm inclined to disagree and say match what VS Code does? 🤷‍♂️

Reply:
do what VS Code did, then